File: android/src/main/kotlin/com/timerly/timerlyplugin/models/CreateForegroundServiceRequest.kt
package com.timerly.timerlyplugin.models

import java.io.Serializable

data class NotificationActionButton(val notificationId: Int, val name: String, val command: String) : Serializable

data class CreateForegroundServiceRequest(val serviceId: Int = 0, val title: String, val message: String, val bigContentTitle: String?, val bigContentMessage: String?, val autoCancel: Boolean? = false, val priority: Int? = 4, val setFullScreenIntent: Boolean? = false,
                                          val channelName: String, val channelId: String, val actionButtons: List<NotificationActionButton>? = listOf(), val channelPriority: Int) : Serializable


data class RemoveNotificationRequest(val notificationId: Int) : Serializable

data class GenericRequest1(val id: Int)

File: lib/models/CreateForegroundServiceRequest.dart
import 'package:meta/meta.dart';
import 'package:timerly_plugin/models/NotificaitonActionButton.dart';

enum notificationPriority {
  IMPORTANCE_NONE,
  IMPORTANCE_MIN,
  IMPORTANCE_LOW,
  IMPORTANCE_DEFAULT,
  IMPORTANCE_HIGH,
  IMPORTANCE_MAX
}

class CreateForegroundServiceRequest {
  int serviceId;
  String title;
  String message;
  String bigContentTitle;
  String bigContentMessage;
  bool autoCancel;
  int priority;
  bool setFullScreenIntent;
  String channelName;
  String channelId;
  List<NotificationActionButton> actionButtons;

  CreateForegroundServiceRequest(
      @required this.serviceId,
      @required this.title,
      @required this.message,
      this.bigContentTitle,
      this.bigContentMessage,
      this.autoCancel,
      this.priority,
      this.setFullScreenIntent,
      this.channelName,
      this.channelId,
      this.actionButtons);

  CreateForegroundServiceRequest.fromJson(Map<String, dynamic> json)
      : serviceId = json['serviceId'],
        title = json['title'],
        message = json['message'],
        bigContentTitle = json['bigContentTitle'],
        bigContentMessage = json['bigContentMessage'],
        autoCancel = json['autoCancel'],
        priority = json['priority'],
        setFullScreenIntent = json['setFullScreenIntent'],
        channelName = json['channelName'],
        channelId = json['channelId'],
        actionButtons = json['actionButtons']
            .map((i) => NotificationActionButton.fromJson(i))
            .toList();

  Map<String, dynamic> toJson() => {
        'serviceId': serviceId,
        'title': title,
        'message': message,
        'bigContentTitle': bigContentTitle,
        'bigContentMessage': bigContentMessage,
        'autoCancel': autoCancel,
        'priority': priority,
        'setFullScreenIntent': setFullScreenIntent,
        'channelName': channelName,
        'channelId': channelId,
        'actionButtons': actionButtons.map((f) => f.toJson()).toList(),
      };
}
